src/EditorFeatures/CSharp/InlineRename/CSharpEditorInlineRenameService.cs

```diff
@@ -18,4 +26,105 @@ internal sealed class CSharpEditorInlineRenameService(
     [ImportMany] IEnumerable<IRefactorNotifyService> refactorNotifyServices,
     IGlobalOptionService globalOptions) : AbstractEditorInlineRenameService(refactorNotifyServices, globalOptions)
 {
+    protected override async Task<ImmutableDictionary<string, ImmutableArray<string>>> GetRenameContextCoreAsync(IInlineRenameInfo inlineRenameInfo, IInlineRenameLocationSet inlineRenameLocationSet, CancellationToken cancellationToken)
+    {
+        var seen = PooledHashSet<TextSpan>.GetInstance();
+        var definitions = ArrayBuilder<string>.GetInstance();
+        var references = ArrayBuilder<string>.GetInstance();
+        var docComments = ArrayBuilder<string>.GetInstance();
+
+        foreach (var renameDefinition in inlineRenameInfo.DefinitionLocations)
+        {
+            var containingStatementOrDeclarationSpan =
+                await renameDefinition.Document.TryGetSurroundingNodeSpanAsync<StatementSyntax>(renameDefinition.SourceSpan, cancellationToken).ConfigureAwait(false) ??
+                await renameDefinition.Document.TryGetSurroundingNodeSpanAsync<MemberDeclarationSyntax>(renameDefinition.SourceSpan, cancellationToken).ConfigureAwait(false);
+
+            var documentText = await renameDefinition.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            if (documentText is null)
+            {
+                continue;
+            }
+
+            var symbolService = renameDefinition.Document.GetLanguageService<IGoToDefinitionSymbolService>();
+            if (symbolService is not null)
+            {
+                var textSpan = inlineRenameInfo.TriggerSpan;
+                var (symbol, _, _) = await symbolService.GetSymbolProjectAndBoundSpanAsync(
```

Comment:
we really don't like doing this.  it makes this work lang specific (it will only work for C#/VB).  

Instead, we should make it so that whatever data is neeeded can be optionally returned by some non-lang-specific ILanguageService that JS/TS/XAML/F# can all implement.

Reply:
ok.  this was already designed to let the other langs play.  but we should at least expose this for C#/VB.  Our policy is that we do not specialize for C# *unless* it's a C#-specific feature (like for a literal lang feature that VB doesn't have).  Otherwise, we light up both.

Reply:
as there is practically nothing lang-specific here, that should be very easy to do (and we can help with that).

Reply:
Well now I think it is because the copied **`[CSharpRenameContextHelper.cs](https://github.com/dotnet/roslyn/pull/73873/files#diff-5d446a8c6ebf9f7cfce4e0ff445c83babf4e658cc97c3c989d81bc330dc5e41d)`** only provide C# implementation.